### ISSUE TITLE
possibility to use in Linux

### ISSUE THREAD

**Issue by toomuchspace** [cfcurtis/sewingutils#27]
Hi!
Is there an opportunity to use your tool in Linux?
And if not yet, is it planned?
Greets from Germany

**Comment by cfcurtis** [comment]
All of the Python libraries are cross platform and compatible with Linux, but I have not built an executable. At the moment I don't have plans to do so, but I think I've seen people run the Python script successfully.

**Comment by mara004** [comment]
Yes, it is easily possible to use pdfsticher on Linux, I am doing so ;)
You can simply install dependencies (wxwidgets, pikepdf) via pip3 and run the pdfsticher.py script (or tile_pages.py for the command-line interface)

**Comment by Triffen** [comment]
Yes, I'm using it on Linux too. Make sure you open it with python3 as a different one might be default for your system (mine is 2.8 I believe).

**Comment by cfcurtis** [comment]
Look into appimage + Ubuntu package manager

**Comment by mara004** [comment]
Maybe it would be an idea to send packaging requests to distributions, e. g. [this](https://wiki.debian.org/RFP) for Debian?

**Comment by cfcurtis** [comment]
I've been playing around with building a binary on linux and for some reason the core wxpython shared object library is huge, like 120 MB instead of the 20ish that I get on mac/windows. I'm using the wxpython wheel at https://extras.wxpython.org/wxPython4/extras/linux/gtk3/ubuntu-20.04/wxPython-4.1.1-cp38-cp38-linux_x86_64.whl for an Ubuntu 20.04 install. I'll keep looking at it, but for now that's kind of an unreasonable file size.

@mara004 I just saw your comment about packaging for distributions. This would be a good thing to look into and might help to make the wx library a dependency instead of part of PDFstitcher.

**Comment by mara004** [comment]
Interesting. I didn't even know there is a wheel - when I installed wxpython from PyPI, it needed to be built from source. I also don't really have a clue why the binary is that large. My first guess was that it might include its own full gtk build, but I do not know.
From looking into the wheel the largest binary is "_core.cpython-38-x86_64-linux-gnu.so" with about 70 MiB

> and might help to make the wx library a dependency instead of part of PDFstitcher.

Yes, I think on Linux it is still the best way to use distribution repositories as much as possible, to avoid installing the same libraries multiple times in different binaries.
Concerning Ubuntu 20.04, it might also be possible to create a ppa (personal package archive at launchpad.net), so that we do not have to wait for newer ubuntu versions to include PDFStitcher in the native repositories.

**Comment by Symbian9** [comment]
There are few utility tools which can help prepare Python apps for AppImage creation:

- https://github.com/linuxdeploy/linuxdeploy-plugin-conda
- https://github.com/niess/linuxdeploy-plugin-python

But do not hesitate to ask for help with AppImage creation:

- https://docs.appimage.org/contact.html

**Comment by nbenitez** [comment]
Hi Charlotte,

[Flathub](https://flathub.org/) is a build and distribution platform for Linux apps, packaged in
a way that makes them portable and running on all major distributions.
Anyone can submit an app (we're not doing it for profit), but one of the
requirements is to reach out to the application's authors if they would
like to help with maintenance and thus making that distribution channel
considered official.

I'm volunteering to make a flatpak for PDF Stitcher to be published in https://flathub.org/.


The usual maintenance for new releases is just changing version string and
the file checksum. The rest is done by flathub's build infraestructure.

It's understandable if you can't commit to maintaining it, I can do the new
version updates as I'm the original packager and interested in having
PDFStitcher on flathub, but we'd like to know **if you are not against
distributing it on Flathub in general**.

Thank you,
Nelson

**Comment by cfcurtis** [comment]
@nbenitez Thank you for the offer, I would be happy to make PDFStitcher available on flathub. 

As you can see my repo is pretty disorganized at the moment, and part of the reason I moved things around last month was to create a better structure for distribution. In particular, I plan to add a proper setup script and a CI workflow so that Windows/Mac/Linux binaries can be bundled using GitHub actions. Unfortunately I started a new job last summer that has been taking all of my time, but this serves as a great reminder/push to get it done.

**Comment by nbenitez** [comment]
> @nbenitez Thank you for the offer, I would be happy to make PDFStitcher available on flathub.

Nice. Good news PDFStitcher is already published on Flathub:
https://flathub.org/apps/details/com.github.cfcurtis.pdfstitcher

So now it's easy to install from any Linux distro.
You were also added as maintainer of the Flathub repo.

Thanks,

**Comment by Symbian9** [comment]
>  if you are not againstdistributing it on Flathub in general.

More options are good, but I still want AppImage.

**Comment by cfcurtis** [comment]
> > if you are not againstdistributing it on Flathub in general.
> 
> More options are good, but I still want AppImage.

I'll keep this issue open. It looks like AppImage would work best if I get it on pypi first, which I've been working on (though wxPython is proving to be a somewhat complicated choice).
